poetry.toml

```diff
@@ -0,0 +1,2 @@
+[virtualenvs]
+in-project = true
```

Comment:
Although we could specify this as a configuration, it isn't really required to run Gaphor. If someone prefers to have all of their virtualenvs managed in the cache-dir for Poetry, that is a fine preference, and I'm not sure if we should override their preference.

Reply:
I'm indeed also a person who generally keeps all virtualenvs in a single location :), but since gaphor does not build without the virtualenv being locally available, and configuring the virtualenv as local is therefore a necessary step to configure the dev environment (e.g. https://docs.gaphor.org/en/latest/macos.html#development-environment). As such, I thought this approach makes life easier. 

Nonetheless, I understand there are use cases in which this is not necessary so I'm happy to revert this. In that case, I would like to add a brief explanation on the use cases for which `poetry config virtualenvs.in-project true`is needed and add the `--local` option to cater to people like me.

Reply:
Hi @teunhoevenaars, I just ran:

```
rm -r .venv
poetry config virtualenvs.in-project false
poetry install
Creating virtualenv gaphor-bY5AB7R2-py3.12 in /home/dan/.cache/pypoetry/virtualenvs
Installing dependencies from lock file
...
poetry run gaphor
```

Gaphor ran fine for me, which is what I was expecting. Are you having issues with running Gaphor like that?

Reply:
Thanks for checking @danyeaw! And yes, that indeed also works fine for me. The use case where it doesn't work is if I want to run
```
poetry install --with packaging
poetry run poe package
```
That's the use case that I referred to when I wrote 'use case for which [...] is needed'. Sorry for the confusion. I'll revert this change back tomorrow and make an alternative suggestion.

Reply:
Interesting, thanks for pointing that out, I didn't realize poe required a local virtual environment. We should report that as a bug to poe library.

Reply:
After reading into the poethepoet documentation (https://poethepoet.natn.io/global_options.html#change-the-executor-type) I believe this was not a bug of the poe library, but an issue with the gaphor pyproject.toml where the poe executor was changed from the default (auto, where it uses the poetry managed environment) to a hard-coded, but default (local), `virtualenv` location. Removing that resolved the issue for me, after which I see no longer any need to specify the location of the poetry virtualenv. 

However, the fact that this config existed means that someone put some thought into this so I'm slightly worried that I've broken something somewhere. If so, I'm guessing that would come up in one of the CI tasks..